### ISSUE TITLE
(User page crashes) Custom Profile Icon

### ISSUE THREAD

**Issue by jaconfurz** [XorTroll/Goldleaf#273]
I tried setting a custom profile icon through goldleaf. But now my profile picture is white. When I try to tap it, my switch crashes. When I try to add a new user, the switch crashes. When I try to change my other account's default picture, it crashes before the window pop-ups. It's like the account service is not working. I can't play any games becaus it crashes before the select user pane pops up. 

Help would be greatly appreciated.

**Comment by XorTroll** [comment]
Goldleaf version?
Please try to use the issue templates to avoid questions like this.

**Comment by jaconfurz** [comment]
My sincere apologies for lack of information in my previous post:

Something else I noticed, if I open Checkpoint, I can see my new profile picture. Though it's definitely not 256 x 256, which is probably why my user page keeps crashing. Is there a way to reset this through goldleaf? Or force it back to a on-device default picture? 

I'm using Goldleaf 0.4
My nintendo switch is version 6.2.0
I ran it using SX OS

**Comment by XorTroll** [comment]
0.4 doesn't do any checks on the image. 0.5 checks whether it is a valid image and it's resolution (now 256x256 will be forced, what could be the issue here)

Not sure how would you fix that, can you create and/or delete users? From system settings

**Comment by jaconfurz** [comment]
I just updated to 0.5, wish I knew size validation wasn't available for 0.4.

I believe the image resolution is what's causing the user pane to crash. I can't create, or delete any users. And can't change any profile pictures without crashing. I also can't load any game without a crash. 

Is there any way to have goldleaf change a profile picture without having the select user screen pop up? This would solve the crash issue. Or even have an emergency picture reset. Where it resets all accounts to, like the first default picture? 

MrCheeze446 from issue  #196 had this exact same problem.

**Comment by Thetoto** [comment]
https://github.com/J-D-K/Avatool
Maybe you can try this homebrew, it doesn't show the user profile pop up

**Comment by jaconfurz** [comment]
Avatool was my next go to, to try and fix this issue. But it crashes after I press A to continue.

I get the following message:
Attempting to shit down account... Succeeded
Attempting to mount 0x8000000010... Succeeded

Then it immediately crashes.

**Comment by jaconfurz** [comment]
Bug: User Applet Crashes after custom Icon

Bug description: When I set custom icon through goldleaf, the profile picture was set to a 1920x1080 image (can see updated profile picture in Checkpoint). Though in user page, profile icon is white. And my switch crashes whenever I tap to change it, or whenever the select user page pops up (like when loading a game). I guess the switch is designed to now show profile icons larger than 256 * 256, so causes the user applet to crash whenever it is called. 

Your setup
Console firmware: 6.2.0
CFW with version: SXOS 2.5.3 Stable. Also tried on Atmosphere 0.8.5 with sig patches. 
Goldleaf version: 0.4

Bug reproduction: Set profile picture as anything besides 256 * 256. User page crashes as switch can't display larger images. And can't use goldleaf to revert back as the switch crashes before the "select a user" window pops up. 

Additional information
If possible. Have a way to define a user without having the select a user window pop up to change profile icon. Or have an emergency profile picture reset, where it sets all users to a default profile picture.

**Comment by jaconfurz** [comment]
Edit: I was able to get this fixed by using Avatool.

**Comment by teamalpha5441** [comment]
Is there any other solution to this problem? Avatool also crashes for me.

**Comment by Abrar1233** [comment]
hey i have the same problem i was wondering how you solved it because avatool also crashes for me

**Comment by teamalpha5441** [comment]
I wrote a little tool which should have just set my avatar to a known good system one, but the actual syscall crashed the system as well, which leads me to believe that fixing the avatar by just setting it new the same way is not possible (take this statement with a grain of salt, I'm a newbie with switch homebrew).

In the end I just backed up my game saves and completely reinstalled the emuMMC. Its a pain in the ass just because of setting a bad avatar, but I couldn't find another working way.

Maybe a 'simple' check could be implemented (I think) here:
https://github.com/XorTroll/Goldleaf/blob/641010bc592fcff2a1a7b533330420049acd188e/Goldleaf/Source/acc/acc_UserSelection.cpp#L134
This would require loading of the image though.
